include/alpaka/core/UniqueId.hpp

```diff
@@ -108,7 +108,7 @@ namespace alpaka
             auto constexpr reader(
                 float,
                 flag<N>,
-                std::size_t R = reader(std::size_t{0u}, flag<N-1u>{}))
+                std::size_t R = reader(std::size_t(0), flag<N-1u>()))
```

Comment:
What is the advantage of using paranthesis versus the brackets?
There is a semantic difference. Your version is a cast while my version fails compilation if the value is not exactly representable by the given type. For the flag constructor there is no difference.